aidial_adapter_bedrock/llm/model/claude/v3/adapter.py

```diff
@@ -62,92 +81,158 @@ async def on_stream_event(self, event: MessageStreamEvent):
 
 
-class ChatParams(TypedDict):
-    max_tokens: int
-    stop_sequences: Union[List[str], NotGiven]
-    system: Union[str, NotGiven]
-    temperature: Union[float, NotGiven]
-    top_p: Union[float, NotGiven]
-    tools: Union[List[ToolParam], NotGiven]
+# NOTE: it's not pydantic BaseModel, because
+# ClaudeMessage.content is of Iterable type and
+# pydantic automatically converts lists into
+# list iterators following the type.
+# See https://github.com/anthropics/anthropic-sdk-python/issues/656 for details.
+@dataclass
+class ClaudeRequest:
+    params: ClaudeParameters
+    messages: List[ClaudeMessage]
 
 
 class Adapter(ChatCompletionAdapter):
-    model: str
+    deployment: Claude3Deployment
     storage: Optional[FileStorage]
     client: AsyncAnthropicBedrock
 
-    async def chat(
-        self,
-        consumer: Consumer,
-        params: ModelParameters,
-        messages: List[Message],
-    ):
-        if params.max_prompt_tokens is not None:
-            raise ValidationError(
-                "max_prompt_tokens request parameter is not supported"
-            )
-
+    async def _prepare_claude_request(
+        self, params: DialParameters, messages: List[DialMessage]
+    ) -> ClaudeRequest:
         if len(messages) == 0:
             raise ValidationError("List of messages must not be empty")
 
         tools = NOT_GIVEN
-        tools_mode = None
+        tool_choice: ToolChoice | NotGiven = NOT_GIVEN
         if params.tool_config is not None:
             tools = [
                 to_claude_tool_config(tool_function)
                 for tool_function in params.tool_config.functions
             ]
-            tools_mode = params.tool_config.tools_mode
+            tool_choice = (
+                {"type": "any"}
+                if params.tool_config.required
+                else {"type": "auto"}
+            )
 
         parsed_messages = [
-            process_with_tools(parse_dial_message(m), tools_mode)
+            process_with_tools(parse_dial_message(m), params.tools_mode)
             for m in messages
         ]
 
-        prompt, claude_messages = await to_claude_messages(
+        system_prompt, claude_messages = await to_claude_messages(
             parsed_messages, self.storage
         )
 
-        completion_params = ChatParams(
+        claude_params = ClaudeParameters(
             max_tokens=params.max_tokens or DEFAULT_MAX_TOKENS_ANTHROPIC,
             stop_sequences=params.stop,
-            system=prompt or NOT_GIVEN,
+            system=system_prompt or NOT_GIVEN,
             temperature=(
                 NOT_GIVEN
                 if params.temperature is None
                 else params.temperature / 2
             ),
             top_p=params.top_p or NOT_GIVEN,
             tools=tools,
+            tool_choice=tool_choice,
+        )
+
+        return ClaudeRequest(params=claude_params, messages=claude_messages)
+
+    async def _computed_discarded_messages(
```

Comment:
minor: since it's a function, we need to use `compute` not `computed`

Reply:
Or does this naming was chosen to distinct it from `compute_discarded_messages` in same class?

Reply:
Good catch, it's a typo. Fixed.